### PR TITLE
LPS-84617 form deserializers, incomplete draft needs review

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LPS-84617, cc @jeyvison can you review this incomplete draft and also answer my questions at the end of the article? Let me know if you'd like to review in a different format, or if I can send you Pull Requests directly.

### CODE REVIEW

File: develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Form Serialization with the DDM IO API
```
Comment (rbohl) [inline develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown]: cc @jeyvison this is the beginning of my draft markdown article.

**Comment by rbohl** [comment]
@jeyvison, bump :)

File: develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    DDMFormSerializerSerializeResponse ddmFormSerializerSerializeResponse =
+    ddmFormSerializer.serialize(builder.build());
+
+    return ddmFormSerializerSerializeResponse.getContent();
+
+In addition to reviewing what I have here, there are some additional questions:
+
+1.  When does this serialization code get called? After a form save?
+2.  Is there configuration in the forms UI to control which serializer is used? Or is it entirely backend code?
+3.  If it's all backend, are developers expected to do something so that their serializer replaces the default JSON serializer?
```
Comment (jeyvison) [inline develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown]: When doing things only in the backend the developer should replace the type of the storage type to the desired format.

File: develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+4.  Is there anything we should tell developers about the deserialization process?
```
Comment (jeyvison) [inline develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown]: Huumm, not that I can think of.

File: develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+5.  Are there any limitations for what formats will work?
```
Comment (jeyvison) [inline develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown]: Any format that can be stored in the database as a string should work.

File: develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+    DDMFormSerializerSerializeResponse ddmFormSerializerSerializeResponse =
+    ddmFormSerializer.serialize(builder.build());
+
+    return ddmFormSerializerSerializeResponse.getContent();
+
+In addition to reviewing what I have here, there are some additional questions:
+
+1.  When does this serialization code get called? After a form save?
+2.  Is there configuration in the forms UI to control which serializer is used? Or is it entirely backend code?
```
[the remark this answers is not in the document]
Comment (rbohl) [reply develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown]: @jeyvison 
I thought that was for the form value storage, not the form itself? Or is this something new in 7.2?

**Comment by rbohl** [comment]
@jeyvison If you create a form serializer, you *must* create a deserializer as well, right?

**Comment by rbohl** [comment]
Please see my most recent commit and let me know what you think. Thanks!

**Comment by jeyvison** [comment]
> @jeyvison If you create a form serializer, you _must_ create a deserializer as well, right?

Yes, exactly.

File: develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown (hunk 5 above)
[the remark this answers is not in the document]
Comment (jeyvison) [reply develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown]: Yeah, my bad. Just confirmed it with Leo. It's only for storage. The Serializer and Deserializer can be only called by backend(from inside the Stora adapter , for example)

**Comment by rbohl** [comment]
Please see my most recent commit and let me know what you think. Thanks!

**Comment by rbohl** [comment]
@jeyvison, another question: we should give the developer an idea of where they'll want to call the serializer o deserializer. So where do they need to call the serialization code or deserialization code? And where does Liferay/DDM call their serialization/deserialization code?

**Comment by jeyvison** [comment]
> Please see my most recent commit and let me know what you think. Thanks!

Already saw it, forgot to tell you XD LGTM :)

> @jeyvison, another question: we should give the developer an idea of where they'll want to call the serializer o deserializer. So where do they need to call the serialization code or deserialization code? And where does Liferay/DDM call their serialization/deserialization code?

They should call it inside their storage adapters.

**Comment by rbohl** [comment]
Oops, sorry for the duplicated comment!

**Comment by rbohl** [comment]
Forwarded: https://github.com/sez11a/liferay-docs/pull/3834

**Pull request closed without merging**